src/components/TodoItem/TodoItem.tsx

```diff
@@ -0,0 +1,143 @@
+import React, { useRef, useState, useEffect } from 'react';
+import classNames from 'classnames';
+import { useTodos } from '../../TodosContext';
+import { Todo } from '../../types/Todo';
+
+type Props = {
+  todo: Todo;
+};
+
+export const TodoItem: React.FC<Props> = ({ todo }) => {
+  const {
+    tempTodos,
+    handleUpdateTodo,
+    handleDeleteTodo,
+    handlerTitleFieldFocused,
+  } = useTodos();
+
+  const [isEditing, setIsEditing] = useState(false);
+  const [newTitle, setNewTitle] = useState(todo.title);
+  const trimmedNewTitle = newTitle.trim();
+
+  const todoElementRef = useRef<HTMLInputElement | null>(null);
+
+  useEffect(() => {
+    if (isEditing) {
+      todoElementRef.current?.focus();
+    }
+  }, [isEditing]);
+
+  const handleCheckboxChange = () => {
+    handleUpdateTodo({ ...todo, completed: !todo.completed });
+  };
+
+  const handleCancelEdit = () => {
+    setIsEditing(false);
+    setNewTitle(todo.title);
+  };
+
+  const handleSaveEdit = async () => {
+    handlerTitleFieldFocused(false);
+
+    if (trimmedNewTitle === '') {
```

Comment:
```suggestion
    if (!trimmedNewTitle) {
```

Reply:
it doesn't work

Reply:
> it doesn't work

It works, https://developer.mozilla.org/en-US/docs/Glossary/Falsy

Reply:
@AndriiKuz in such cases, it is worth confirming with a screenshot or a short video that it does not work, because it should work, this is a feature of the JavaScript language)